staging/src/k8s.io/apiserver/pkg/server/filters/maxinflight.go

```diff
@@ -141,21 +155,18 @@ func WithMaxInFlightLimit(
 
 			select {
 			case c <- true:
-				var mutatingLen, readOnlyLen int
 				if isMutatingRequest {
-					mutatingLen = len(mutatingChan)
+					watermark.recordMutating(len(c))
 				} else {
-					readOnlyLen = len(nonMutatingChan)
+					watermark.recordReadOnly(len(c))
 				}
-
 				defer func() {
 					<-c
 					if isMutatingRequest {
-						watermark.recordMutating(mutatingLen)
+						watermark.recordMutating(len(c))
 					} else {
-						watermark.recordReadOnly(readOnlyLen)
+						watermark.recordReadOnly(len(c))
```

Comment:
Is this purposeful, to record the queue length twice? If so, would you mind adding a comment on why this is necessary?

Reply:
Yes, it is purposeful.  The reported metrics have been extended, it is not just the high water mark now, so the lower level needs to be noted as well as the higher.  Added a comment.